staging/src/kubevirt.io/api/export/v1alpha1/types.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VirtualMachineExport defines the operation of exporting a VM source
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineExport struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineExportSpec `json:"spec"`
+
+	// +optional
+	Status *VirtualMachineExportStatus `json:"status,omitempty"`
+}
+
+// VirtualMachineExportList is a list of VirtualMachineExport resources
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineExportList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	// +listType=atomic
+	Items []VirtualMachineExport `json:"items"`
+}
+
+// VirtualMachineExportSpec is the spec for a VirtualMachineExport resource
+type VirtualMachineExportSpec struct {
+	Source corev1.TypedLocalObjectReference `json:"source"`
+
+	// TokenSecretRef is the name of the secret that contains the token used by the export server pod
+	TokenSecretRef string `json:"tokenSecretRef"`
+}
+
+// VirtualMachineExportPhase is the current phase of the VirtualMachineExport
+type VirtualMachineExportPhase string
+
+const (
+	// Pending means the Virtual Machine export is pending
+	Pending VirtualMachineExportPhase = "Pending"
+	// Ready means the Virtual Machine export is ready
+	Ready VirtualMachineExportPhase = "Ready"
+	// Terminated means the Virtual Machine export is terminated and no longer available
+	Terminated VirtualMachineExportPhase = "Terminated"
+)
+
+// VirtualMachineExportStatus is the status for a VirtualMachineExport resource
+type VirtualMachineExportStatus struct {
+	// +optional
+	Phase VirtualMachineExportPhase `json:"phase,omitempty"`
+
+	// +optional
+	Links *VirtualMachineExportLinks `json:"links,omitempty"`
+
+	// +optional
+	ServiceName string `json:"serviceName,omitempty"`
```

Comment:
Its the name of the export service (this would be part of the internal links, since the service name defines the internal fqdn). But I agree I should add some documentation on what it is.